### ISSUE TITLE
suggestion: fn to compute the pairwise distances matrix

### ISSUE THREAD

**Issue by awhillas** [Dicklesworthstone/fast_vector_similarity#1]
Hey,

I wanted to try the different metrics with [silhouette_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_score.html#sklearn.metrics.silhouette_score) as you can pass it a matrix of pairwise distances between vectors. Would be handy if you had a function that produced this since it would be much faster in Rust than python.

btw love you lib!

**Comment by Dicklesworthstone** [comment]
Cool, I will try to add that.

**Comment by Dicklesworthstone** [comment]
Thanks for the feature suggestion! This is still on the backlog but hasn't been implemented yet.

For now, if you need pairwise distance matrices, you could:
1. Use the existing `compute_vector_similarity_stats` in a loop (Python-level parallelization)
2. Use scipy's `pdist` with a custom metric wrapper calling this library

This would be a nice addition for v2 - a function like `compute_pairwise_distance_matrix(vectors: List[List[float]], metric: str) -> np.ndarray` that computes all N*(N-1)/2 unique pairs in parallel using Rayon.

Leaving this open as a valid feature request for future consideration.

**Comment by Dicklesworthstone** [comment]
Closing this stale feature request. If still needed, please open a new issue.
